client/reader/recommend-button/index.tsx

```diff
@@ -0,0 +1,54 @@
+import { Button } from '@wordpress/components';
+import clsx from 'clsx';
+import { useTranslate } from 'i18n-calypso';
+import { useEffect } from 'react';
+import { useRecommendedSite } from 'calypso/landing/subscriptions/hooks/use-recommended-site';
+import { useDispatch, useSelector } from 'calypso/state';
+import { getCurrentUserName } from 'calypso/state/current-user/selectors';
+import { requestRecommendedBlogsListItems } from 'calypso/state/reader/lists/actions';
+import {
+	hasRequestedUserRecommendedBlogs,
+	isRequestingUserRecommendedBlogs,
+} from 'calypso/state/reader/lists/selectors';
+import ReaderFollowFeedIcon from '../components/icons/follow-feed-icon';
+import ReaderFollowingFeedIcon from '../components/icons/following-feed-icon';
+import type { Feed } from 'calypso/state/data-layer/wpcom/read/feed/types';
+import './style.scss';
+
+interface Props {
+	feedId: Feed[ 'feed_ID' ];
+}
+
+export const RecommendButton = ( { feedId }: Props ) => {
+	const { isRecommended, toggleRecommended } = useRecommendedSite( feedId );
```

Comment:
We could potentially set this up as a property to pass into this component instead, similar to how the Follow Button works for isFollowing IIRC.

Then if we `useRecommendedSite` where we add this component (in reader feed header for example), we could condition showing this recommend component as something like `(isSubscribed || isRecommended) && <RecommendButton />`.

Alternatively, we could add a subscription check in here as well and do an early return if they are not subscribed and aren't currently recommending the item. The former might be more flexible though.

Reply:
> but I didn't like the result either.

It seems reasonable for the original intent! I suppose my comment is more around the late shift to want to control this buttons visibility more with whether or not the user is subscribed.